### DISCUSSION TITLE
multibyte characters in the post are garbled

### DISCUSSION THREAD

**Discussion by NikaidoKota** [selenide/selenide#2204] [Q&A]
Hi!

I use Selenide for UI testing of pages written in Japanese.
(Java8, Selenide 5.25.0)

When "Windows-31J" is selected for the page encoding, multibyte characters in POST methods are garbled.
Specifically, the following is declared in jsp.
`<% page language="java" contentType="text/html; charset=Windows-31J" pageEncoding="Windows-31J" %>`

I would like to eliminate garbled pages without changing this declaration.
I think I need to match the encoding of the client to "Windows-31J".
I don't know how to set this encoding, so any help would be appreciated.

Thank you.

**Comment by asolntsev** [comment]
@NikaidoKota Sorry, I did't understand what exactly is not working. 
I tried to open a page in Japanese with Selenide - it works well: https://github.com/selenide/selenide/commit/f52ed442541374ff7be7880625b64abdd3795e00

**Reply by NikaidoKota** [reply]
I am using Selenide and Chrome Webdriver to test a web application coded in Java/JSP.
The character code is "Windows-31J".
The web app has a form for inputting multibyte characters, and by pressing the "Decide" button and submitting, the contents of the form are POSTed.
The form then verifies that the multibyte string has been entered with the appropriate number of characters, but it seems that the string entered at the time of submit is recognised as a UTF-8 string, so the multibyte characters are garbled and a verification error occurs.
If the screen is opened manually and input with the keyboard without passing through the Webdriver, the "Windows-31J" character string can be submitted correctly and no error occurs.
Therefore, I suspect that for some reason the character encoding is fixed to UTF-8 when using Webdriver.

**Reply by asolntsev** [reply]
Hm... it might be a chromedriver bug, for example...

Where can I try such a site?
In other words, how i could easily reproduce the problem?
